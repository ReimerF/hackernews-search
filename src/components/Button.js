import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Button extends Component {
    render () {
        const { children, className, onClick} = this.props;
        return (
            <button
                className={className}
                onClick={onClick}
                type="button">
                {children}
            </button>
        )
    }
}

Button.propTypes ={
    onClick: PropTypes.func.isRequired,
    classname: PropTypes.string,
    children: PropTypes.node.isRequired,
};

Button.defaultProps = {
    classname: '',
}

export default Button;