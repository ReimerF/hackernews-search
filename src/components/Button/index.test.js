import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Button from "./index";

describe("Button", () => {
  it("renders Button without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Button onClick={jest.fn()}>Testing happening here</Button>,
      div
    );
  });

  test("Button has a valid snapshot", () => {
    const component = renderer.create(
      <Button onClick={jest.fn()}>Testing happening here</Button>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
