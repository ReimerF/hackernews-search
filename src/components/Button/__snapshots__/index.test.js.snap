// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button has a valid snapshot 1`] = `
<button
  onClick={[MockFunction]}
  type="button"
>
  Testing happening here
</button>
`;
